spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -1540,8 +1540,10 @@ protected void executeListener(Channel channel, Object data) {
 			observation = Observation.NOOP;
 		}
 		else {
+			Message message = (Message) data;
 			observation = RabbitListenerObservation.LISTENER_OBSERVATION.observation(registry,
-						new RabbitMessageReceiverContext((Message) data, getListenerId()))
+						new RabbitMessageReceiverContext(message, getListenerId(),
+								message.getMessageProperties().getConsumerQueue()))
```

Comment:
May we just do this in the target `RabbitMessageReceiverContext`?
We already provide a `Message` over there...

Reply:
But it's not available in `getContextualName()`.

Reply:
Oh; sorry - I see what you mean.